### ISSUE TITLE
Intended behavior for 1.*.1

### ISSUE THREAD

**Issue by dtolnay** [steveklabnik/semver-parser#62]
My expectation is that `1.*.1` is not a meaningful version req and should fail to parse. However I don't have a spec to point to so I just wanted to confirm whether semver-parser's current behavior on this case is intended.

```rust
fn main() {
    println!("{:#?}", "1.*.1".parse::<semver_parser::RangeSet>().unwrap());
}
```

As of the most recent release (0.10.2) semver-parser parses it as follows, i.e. `>=1.0.0, <2.0.0`.

```console
RangeSet {
    ranges: [
        Range {
            comparator_set: [
                Comparator {
                    op: Gte,
                    major: 1,
                    minor: 0,
                    patch: 0,
                    pre: [],
                },
                Comparator {
                    op: Lt,
                    major: 2,
                    minor: 0,
                    patch: 0,
                    pre: [],
                },
            ],
            compat: Cargo,
        },
    ],
    compat: Cargo,
}
```

**Comment by steveklabnik** [comment]
It should fail to parse, yes. If there are existing packages with versions on crates.io that are incorrect, we could add them into https://github.com/steveklabnik/semver/blob/051c39c50d565c597be65c4496597c7d16a514d9/src/version_req.rs#L323-L332 to keep compatibility, but the intention isn't for this sort of version to parse properly.

https://github.com/semver/semver/pull/584 is the spec I intend to follow once I get around to actually finally double checking everything and getting it in.

**Comment by dtolnay** [comment]
This is fixed in 1.0.0.

```rust
fn main() {
    println!("{}", semver::VersionReq::parse("1.*.1").unwrap_err());
}
```

```console
unexpected character after wildcard in version req
```
